python/perspective/setup.py

```diff
@@ -146,31 +152,41 @@ def build_extension_cmake(self, ext):
             build_args += ['--', '/m']
         else:
             cmake_args += ['-DCMAKE_BUILD_TYPE=' + cfg]
-            build_args += ['--', '-j2' if os.environ.get('DOCKER', '') else '-j4']
+            build_args += ['--', '-j2' if os.environ.get('DOCKER', '') else '-j{}'.format(CPU_COUNT)]
 
         env = os.environ.copy()
         env['PSP_ENABLE_PYTHON'] = '1'
         env["PYTHONPATH"] = os.path.pathsep.join((os.path.join(os.path.dirname(os.__file__), 'site-packages'), os.path.dirname(os.__file__)))
 
         if not os.path.exists(self.build_temp):
             os.makedirs(self.build_temp)
-
         try:
-            subprocess.check_output([self.cmake_cmd, os.path.abspath(ext.sourcedir)] + cmake_args, cwd=self.build_temp, env=env, stderr=subprocess.STDOUT)
-            subprocess.check_call([self.cmake_cmd, '--build', '.'] + build_args, cwd=self.build_temp, env=env, stderr=subprocess.STDOUT)
+            out1 = subprocess.check_output([self.cmake_cmd, os.path.abspath(ext.sourcedir)] + cmake_args, cwd=self.build_temp, env=env, stderr=subprocess.STDOUT)
```

Comment:
`check_output` has really shitty user experience when building from the console - errors lack highlighting and incremental output is suppressed.  If there is no functional reason to change this logging, I would prefer to revert these.

Reply:
I have no preference, but I'm done trial-and-erroring this code, whoever wants this to change can implement that change

Reply:
Not asking for a change - this reverts the current behavior on master for a reasons I don't understand yet.  Can I revert this?

Reply:
The current master raises an exception if the subprocess call fails because out has no attribute decode.